backend/apis/reports/zipcode-report.api.js

```diff
@@ -0,0 +1,58 @@
+const { app, invalidRequest, pool, databaseError } = require("../../server");
+const { checkValid, nullableValidDate } = require("../utils/validation-utils");
+const mysql = require("mysql");
+const _ = require("lodash");
+const ClientCapability = require("twilio/lib/jwt/ClientCapability");
+
+app.get(`/api/reports/client-zipcodes`, (req, res) => {
+  const validationErrors = checkValid(
+    req.query,
+    nullableValidDate("start"),
+    nullableValidDate("end")
+  );
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  const startDate = req.query.start || "2000-01-01T0";
+  const endDate = req.query.end || "3000-01-01T0";
+
+  const sql = mysql.format(
+    `
+      SELECT zip, city, COUNT(*) as clientCount
```

Comment:
By selecting both the zip and city, we will now have multiple rows for the same zip code, in the situations where the zip code doesn't match city boundaries. This might be confusing when reading the report, since the report is a zip code report but would not show you the grand total for a zip code in the situations where there were two cities for the same zip code.

To resolve, we could either show a grand total for each zip code, or we could remove the city from this query so that there's guaranteed to only have one row per zip code. The latter option is simpler to implement and totally fine for CU - they know most of the zip codes and can look up the rest.

Reply:
Good call out. I have removed the city.